include/Task.hpp

```diff
@@ -0,0 +1,15 @@
+//
+// Created by Tianyi Wang on 5/14/23.
+//
+
+#ifndef DD_EVAL_TASK_HPP
+#define DD_EVAL_TASK_HPP
+
+#endif // DD_EVAL_TASK_HPP
+
+#include "Result.hpp"
+
+class Task {
+public:
+  virtual Result execute() = 0;
```

Comment:
I would not have expected the `Task` to have an `execute` function. That's the job of the `Executor`.
The `Task` just describes an instance to be benchmarked.
For Simulation tasks that would be a single circuit, for verification tasks it would be two circuits.
The `Task` class then only needs appropriate constructors and helper functions.

Reply:
My initial idea was to apply a Command design pattern and have a void execute() function from the invoker (executor) which calls the execute() function in the command (task). This approach although kind of confused me when I was looking at the benchmarking stuff in qmap and thinking about how to implement it. I'll give it some more thought👍